test/node_search.cc

```diff
@@ -191,7 +191,7 @@ void graph_builder::write_tiles(uint8_t level) const {
 
     auto& tile = writer.builder(tile_id);
     node_ids.emplace_back(tile_id.tileid(), level, tile.nodes().size());
-    tile.nodes().emplace_back(std::move(node_builder));
+    tile.nodes().emplace_back(node_builder);
```

Comment:
is it possible that clang knows that the compiler will automatically use rvalue references, kind of like return value optimization?

Reply:
and if it knows that, is it only valid for clang compilers? because our main platform is ubuntu gcc so...